baybe/utils/interval.py

```diff
@@ -41,40 +45,60 @@ class Interval:
     """The upper end of the interval."""
 
     @upper.validator
-    def _validate_order(self, _: Any, value: float):  # noqa: DOC101, DOC103
+    def _validate_order(self, _: Any, upper: float) -> None:  # noqa: DOC101, DOC103
         """Validate the order of the interval bounds.
 
         Raises:
             ValueError: If the upper end is not larger than the lower end.
         """
-        if value <= self.lower:
+        if upper <= self.lower:
             raise ValueError(
-                f"The upper interval bound (provided value: {value}) must be larger "
+                f"The upper interval bound (provided value: {upper}) must be larger "
                 f"than the lower bound (provided value: {self.lower})."
             )
 
     @property
-    def is_finite(self):
-        """Check whether the interval is finite."""
-        return np.isfinite(self.lower) and np.isfinite(self.upper)
+    def is_bounded(self) -> bool:
+        """Check if the interval is bounded."""
+        return self.is_left_bounded and self.is_right_bounded
 
     @property
-    def is_bounded(self):
-        """Check whether the interval is bounded."""
-        return np.isfinite(self.lower) or np.isfinite(self.upper)
+    def is_left_bounded(self) -> bool:
+        """Check if the interval is left-bounded."""
+        return np.isfinite(self.lower)
+
+    @property
+    def is_right_bounded(self) -> bool:
+        """Check if the interval is right-bounded."""
+        return np.isfinite(self.upper)
+
+    @property
+    def is_half_bounded(self) -> bool:
+        """Check if the interval is half-bounded."""
+        return self.is_left_bounded ^ self.is_right_bounded
+
+    @property
```

Comment:
we really need this deprecation? imo not. if so, it should list in changelog under deprecatoins too

Reply:
"Need" is perhaps to strong of a statement 😄 But the same is true for the "BayBE" object before the point of open sourcing, and certain other deprecations. There, we said "let's start using a proper deprecation protocol from now on" and that's what I followed. Having it doesn't hurt and the code is already there, hence I'd say let's leave it and thus I resolve. But if you are really against, I'm not blocking. Feel free to shoot a commit that removes it :D Apart from that, the PR is ready from my end, so let me know if anything else needs change 👍🏼 I'll already prepare the update with the current main.